src/main/java/de/tum/in/www1/artemis/service/TutorParticipationService.java

```diff
@@ -113,33 +114,29 @@ public TutorParticipation createNewParticipation(Exercise exercise, User tutor)
     }
 
     /**
-     * Validates tutor feedback.
+     * Validates if tutor feedback matches instructor feedback.
      * Validation rules:
-     * - There should exist a corresponding instructor feedback that references the same object
-     * - If instructor feedback has a grading instruction associated with it, so must the tutor feedback
      * - The feedback should have the same creditCount(score)
+     * - If instructor feedback has a grading instruction associated with it, so must the tutor feedback
+     * - Tutor feedback is not allowed to have negative score without feedback content
      *
      * @return error type if feedback is invalid, `Optional.empty()` otherwise.
      */
-    private Optional<FeedbackCorrectionErrorType> checkTutorFeedbackForErrors(Feedback tutorFeedback, List<Feedback> instructorFeedback) {
-        Optional<Feedback> maybeMatchingInstructorFeedback = instructorFeedback.stream().filter(feedback -> Objects.equals(tutorFeedback.getReference(), feedback.getReference()))
-                .findFirst();
-
-        // In case there is no instructor feedback that is referencing the same element (text/model), return unnecessary feedback.
-        if (maybeMatchingInstructorFeedback.isEmpty()) {
-            return Optional.of(FeedbackCorrectionErrorType.UNNECESSARY_FEEDBACK);
+    private Optional<FeedbackCorrectionErrorType> tutorFeedbackMatchesInstructorFeedback(Feedback tutorFeedback, Feedback instructorFeedback) {
```

Comment:
I personally like the old name more, since it explains the returned value. Checking this method for emptiness (like done in line 172) was not self-explanatory to me, I only understood why you did it after looking up the returned datatype.

Reply:
`checkTutorFeedbackForErrors` is used on line L153, this is another helper method